nomad/server.go

```diff
@@ -1147,11 +1150,15 @@ func (s *Server) setupRPC(tlsWrap tlsutil.RegionWrapper) error {
 }
 
 // setupRpcServer is used to populate an RPC server with endpoints
-func (s *Server) setupRpcServer(server *rpc.Server, ctx *RPCContext) {
-	// Add the static endpoints to the RPC server.
+func (s *Server) setupRpcServer(server *rpc.Server, ctx *RPCContext) error {
 
-	encrypter := NewEncrypter()
+	// Set up the keyring
+	encrypter, err := NewEncrypter(filepath.Join(s.config.DataDir, "server", "keystore"))
```

Comment:
Isn't `server` is added to `config.DataDir` in command/agent/agent.go. https://github.com/hashicorp/nomad/blob/24fab57e0f8b34d21fd69857272fd23c74ad0f05/command/agent/agent.go#L195-L197

Reply:
Yes, good catch!

Reply:
Done in 6eefd9e993f790269c692a1a04e5ccd79f7ec490